single_kernel_mongo/state/charm_state.py

```diff
@@ -278,33 +306,63 @@ def peer_units_data_interfaces(self) -> dict[Unit, DataPeerOtherUnitData]:
             for unit in self.peers_units
         }
 
+    @property
+    def formatted_socket_path(self) -> str:
+        """URL encoded socket path."""
```

Comment:
can you embed some of our expert knowledge here. i.e. this is primarily used for mongos subordinate VM charm in order to improve latency